syft/execution/plan.py

```diff
@@ -384,6 +450,48 @@ def __setattr__(self, name, value):
             for tensor_name, tensor in value.named_tensors():
                 self.__setattr__(f"{name}_{tensor_name}", tensor)
 
+    def validate_input(self, args, kwargs):
+        """
+        Function used in __call__ to validate the input based on the input received at build time.
+
+        Args:
+            args: original args received on __call__.
+            kwargs: original kwargs received on __call__.
+        """
+
+        def extract_type(obj: PlaceHolder) -> str:
+            """
+            Function to extract type on the
+            """
+            for tag in obj.tags:
+                if "type_" in tag:
+                    return tag[6:]
```

Comment:
Should be 5 no ?

Reply:
The actual form of the type is #type_<int>, respecting the convention from the other tags.